tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -704,32 +664,27 @@ bool IsHLSLObjectWithImplicitROMemberAccess(clang::QualType type) {
 }
 
 bool IsHLSLRWNodeInputRecordType(clang::QualType type) {
-  if (const RecordType *RT = type->getAs<RecordType>()) {
-    StringRef name = RT->getDecl()->getName();
-    if (name == "RWDispatchNodeInputRecord" ||
-        name == "RWGroupNodeInputRecords" || name == "RWThreadNodeInputRecord")
-      return true;
-  }
+  if (const HLSLNodeObjectAttr *Attr = getNodeAttr(type))
+    return Attr->getType() == HLSLNodeObjectAttr::RWDispatchNodeInputRecord ||
+           Attr->getType() == HLSLNodeObjectAttr::RWGroupNodeInputRecords ||
+           Attr->getType() == HLSLNodeObjectAttr::RWThreadNodeInputRecord;
```

Comment:
I know you reluctantly changed the above, but for consistency we could do something like this: 

```c++
return (static_cast<uint32_t>(GetNodeIOType(type)) &
          (static_cast<uint32_t>(DXIL::NodeIOFlags::ReadWrite) |  static_cast<uint32_t>(DXIL::NodeIOFlags::Input))) ==
          (static_cast<uint32_t>(DXIL::NodeIOFlags::ReadWrite) |  static_cast<uint32_t>(DXIL::NodeIOFlags::Input));
```